test/pg-cdc/reject-multiple-materializations.td

```diff
@@ -0,0 +1,78 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+#
+# Test that any replication slot can only be materialized once
+#
+
+
+# Insert data pre-snapshot
+$ postgres-execute connection=postgres://postgres:postgres@postgres
+ALTER USER postgres WITH replication;
+DROP SCHEMA IF EXISTS public CASCADE;
+DROP PUBLICATION IF EXISTS mz_source;
+
+CREATE SCHEMA public;
+
+CREATE TABLE t1 (id SERIAL PRIMARY KEY, f1 BOOLEAN);
+ALTER TABLE t1 REPLICA IDENTITY FULL;
+
+CREATE TABLE t2 (id SERIAL PRIMARY KEY, t1_id INT REFERENCES t1(id), name VARCHAR);
+
+INSERT INTO t1(f1) VALUES ('true'),('false');
+
+INSERT INTO t2(t1_id, name) VALUES (1, 'example');
+
+CREATE PUBLICATION mz_source FOR ALL TABLES;
+
+> CREATE SOURCE mz_source
+  FROM POSTGRES CONNECTION 'host=postgres port=5432 user=postgres password=postgres sslmode=require dbname=postgres'
+  PUBLICATION 'mz_source';
+
+> CREATE VIEWS FROM SOURCE mz_source;
+
+> CREATE MATERIALIZED VIEW t1_mat AS
+  SELECT * FROM t1
+
+> SELECT id, f1 FROM t1_mat;
+1 true
+2 false
+
+! CREATE MATERIALIZED VIEW t1_mat_dupe AS
+  SELECT * FROM t1
+Cannot re-materialize source mz_source
+
+! CREATE MATERIALIZED VIEWS FROM SOURCE mz_source (t1 as t1_not_a_dupe, t2 as t2_not_a_dupe)
+Cannot re-materialize source mz_source
+
+> DROP VIEW t1_mat;
+
+> CREATE MATERIALIZED VIEW t1_mat AS SELECT * FROM t1
+
+! SELECT * FROM t1_mat;
+must be dropped and recreated
```

Comment:
Hm, this surprised me. Why isn't `t1_mat` working here since it's the only materialization?

Reply:
Hmm I'm pretty sure that this used to require re-creating the source itself to recreate the publication slot (i.e. the message had to do with `mz_source must be dropped and recreated`) but indeed I can no longer reproduce this.

Is that correct? Shouldn't the replication slot need to be recreated in order for the rematerialized view to work correctly?

Reply:
The source will always recreate the slot when it starts up so that's why it's ok. I think it's ok to ship with this limitation if it's too hard to make the code allow this case